labs/a8.md

```diff
@@ -0,0 +1,736 @@
+of symmetric key crypto is by the XOR function: `a XOR b` is true if and only 
+if a and b are different. Suppose our plaintext is the bitstring  
+`100110100101`, and we want to apply a bitwise XOR with the bitstring 
+`010010100010`, which is our key. The result of this operation is 
+`110100000111`. If we XOR the resulting ciphertext with the key again, we get 
+the original plaintext: `100110100101`. The XOR function is trivially 
+reversible, but military-grade symmetric key cryptography algorithms are much 
+more difficult to reverse. The security of symmetric-key crypto, therefore, is 
+dependent on keeping the encryption key secret.
+
+What is symmetric key crypto useful for? Almost everything: encrypting data in 
+transit (for example, in HTTPS), encrypting data at rest (for example, on your 
+iPhone or Android device, which is "encrypted by default" in recent versions),
+and many other applications. On the iPhone specifically, several layers of
+symmetric crypto are at play: the iPhone's internal storage is encrypted with a
+set of AES keys stored on a chip inside the phone, generated at the factory.
+These keys are in turn encrypted with your PIN. Your pin allows the phone to
+unlock the keys that allow it to decrypt the rest of the filesystem. This is
+also how the iPhone is able to quickly wipe your data in the event your phone is
+stolen: the phone simply deletes the keys that are stored on the internal chip.
+Now, even with your PIN, there are no keys to decrypt and the encrypted data in
+the phone's storage is, for all intents and purposes, irretrievable and
+indistinguishable from random data.
+
+##### Public Key Cryptography
+
+Unlike symmetric key cryptography, there are 2 keys in a public-key 
+cryptosystem, the public key and the private key. As the name suggests, the 
+public key is shared publicly, and this is the means by which other people 
+encrypt data that's meant for you. You use your private key to decrypt this 
+data. As long as no one has your private key, anyone can use your public key to 
+encrypt data and be assured that only you can decrypt it. This is a powerful 
+expansion on the symmetric-key paradigm, as beyond encryption, it also allows 
+for signatures and non-repudiation: a way for someone to verify that the person 
+they are talking to is in fact the person they intend to be talking to, and for 
+someone to prove (without the ability to deny it) that they are who they say 
+they are.
+
+Nowadays, public-key cryptography is synonymous with the RSA algorithm, which 
+was one of the first proven dual-key schemes. You will encounter the RSA 
+algorithm in CS 70 and CS 161 if you plan to take those courses, or already 
+have. In short, the security of RSA depends on the theoretical difficulty of 
+factoring large numbers on conventional computers. This is expected to continue 
+to be a difficult problem until quantum computers become practical.
```

Comment:
A lot of this text could probably get cut out, since we already explained public key crypto in a4. Also they probably know it.